rgw: move compression config into zone placement

cbodley opened the pull request:
This replaces the `rgw_compression_type` config option with an entry in the zone configuration, which can be set with the `--compression=` option to `radosgw-admin zone create` or `zone modify`.

Also adds some new documentation.

cbodley commented:
the new doc can be viewed here: http://docs.ceph.com/docs/wip-rgw-zone-compression/radosgw/compression/

cbodley commented:
@oritwas could you help review please?

cbodley commented:
adding @yehudasa

yehudasa commented:
@cbodley actually, maybe we want to have this as part of the specific placement target config, and not globally per zone?

oritwas commented:
@cbodley , I think making it part of specific placement target config sounds useful.

cbodley commented:
> @cbodley actually, maybe we want to have this as part of the specific placement target config, and not globally per zone?

adding it to placement targets does make sense

i'd prefer not to require admins to edit their zone json to set these, so new radosgw-admin commands to add/remove/edit placement targets would be nice. and the placement target stuff is only documented under `Regions: Set a Region` in config-ref.rst - that stuff really needs to be updated to reflect the new multisite terminology (and probably moved into multisite.rst)

are these changes in scope for the kraken rc? or should the compression doc just say `radosgw-admin zone get` -> `[edit json]` -> `zone set` for now?

cbodley commented:
* moved the `compression_type` field into `RGWZonePlacementInfo`
* added the `--compression` option to the `radosgw-admin zone placement` commands
* moved documentation into a separate pr at https://github.com/ceph/ceph/pull/12244
* marked [DNM] because this includes unmerged commits from https://github.com/ceph/ceph/pull/12242

tested with a custom multisite cluster, with compression enabled thusly:
```
radosgw-admin zone placement modify --rgw-zone na-1 --placement-id=default-placement --compression=random
```

cbodley commented:
rebased after merge of https://github.com/ceph/ceph/pull/12242 and removed [DNM] tag
